chirp/drivers/ft4.py

```diff
@@ -1149,12 +1150,35 @@ def get_duplex(freq):    # auto duplex to real duplex
                 mem.offset = txfreq
             else:
                 mem.offset = int(_mem.offset) * 25000 * freq_offset_factor
+            if self.DUPLEX_OFF_VIA_OFFSET and \
+               regtype in ["memory", "pms"] and mem.duplex in ['+', '-']:
```

Comment:
This is the source of the only error I see. The `regtype in ["memory"...]` bit excludes the special memories from this big-offset handling, which means the special memory test fails to be able to set `duplex=off`, because it just sees `off` turn into `+99MHz` as expected. Why this filter? If I remove the regtype check, all the tests pass for me.